### ISSUE TITLE
NullReferenceException: An exception occurred translating that FullConditionalExpression

### ISSUE THREAD

**Issue by MaceWindu** [agileobjects/ReadableExpressions#105]
```
Ut-oh
An exception occurred translating that FullConditionalExpression. 

Please report this error with the stack trace below using https://github.com/agileobjects/ReadableExpressions/issues/new . 

Thanks! (and sorry about that)
Steve


--------------------------------------------------------------------------------

System.NullReferenceException: Object reference not set to an instance of an object. at
AgileObjects.ReadableExpressions.Translations.FixedValueTranslation.AgileObjects.ReadableExpressions.Translations.ITranslatable.get_TranslationSize() at AgileObjects.ReadableExpressions.Translations.CodeBlockTranslation.CalculateSizes() at
AgileObjects.ReadableExpressions.Translations.CodeBlockTranslation..ctor(ITranslation translation, ITranslationContext context) at
AgileObjects.ReadableExpressions.Extensions.PublicTranslationContextExtensions.GetCodeBlockTranslationFor(ITranslationContext context, Expression expression) at
AgileObjects.ReadableExpressions.Translations.ConditionalTranslation.TernaryTranslation..ctor(ConditionalExpression conditional, ITranslationContext context) at AgileObjects.ReadableExpressions.Translations.ConditionalTranslation.For(ConditionalExpression conditional, ITranslationContext context) at
AgileObjects.ReadableExpressions.Translations.ExpressionTranslation.GetTranslationFor(Expression expression) at AgileObjects.ReadableExpressions.Translations.ExpressionTranslation.GetTranslation() at
AgileObjects.ReadableExpressions.ExpressionExtensions.ToReadableString(Expression expression, Func`2 configuration) at
AgileObjects.ReadableExpressions.Visualizers.ObjectSource.ExpressionVisualizerObjectSource.GetTranslationFor(Object target) at
AgileObjects.ReadableExpressions.Visualizers.ObjectSource.ExpressionVisualizerObjectSource.GetData(Object target, Stream outgoingData, Action`2 serializer)


```

![image](https://user-images.githubusercontent.com/1099238/154803830-8052f714-f15b-4c7a-8ba1-bbb5c6d7a7bf.png)

non-printed parameter is MySqlDecimal struct value (default)

**Comment by SteveWilkes** [comment]
Hi,

Thanks for reporting this!

Could you let me know - how is the default MySqlDecimal value being encoded in the Expression? I've tried a few ways (e.g see [this DotNetFiddle](https://dotnetfiddle.net/cH4hLr)) but can't reproduce the error, which is being thrown by the code trying to translate the default value in a way I wouldn't expect.

Cheers,

Steve

**Comment by MaceWindu** [comment]
looks like this one triggers it for me
```cs
MySqlDecimal val = default;
        var defaultDecimalConstant = Constant(val);
```

**Comment by SteveWilkes** [comment]
Ah yes, that does it - weird, cos if you switch `MySqlDecimal` to `Guid`, it works. I'll look into it...

**Comment by SteveWilkes** [comment]
Looks like `MySqlDecimal.ToString()` will return `null` from its default value (it just [returns its underlying value]((https://github.com/mysql/mysql-connector-net/blob/8.0/MySQL.Data/src/Types/MySqlDecimal.cs))), which it should never do.

I've updated the code to handle nulls returned from a ToString() call anyway - I guess someone could implement their own Extension Expression and return null for their own reasons, so it's better not to fall over.

Thanks again for letting me know about the issue.

**Comment by MaceWindu** [comment]
Actually, `object.ToString` has `string?` signature. A bit annoying, but `null` is valid case

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Object.cs#L39

**Comment by SteveWilkes** [comment]
Weird, Visual Studio seems to have a different opinion:

![image](https://user-images.githubusercontent.com/8459425/154847689-a3d14301-f5dc-4600-bc1d-8286d8a7620c.png)

**Comment by MaceWindu** [comment]
that's because you have `string` as return type (overloads could have more strict return type)

**Comment by SteveWilkes** [comment]
Ahh, yes - good point. Still, if `default(int?).ToString()` returns an empty string instead of null, I think it's at least fair to say there's a convention of doing so.

In any case, the latest code handles nulls :)

**Comment by MaceWindu** [comment]
BTW, if you are interested in more details why ToString is nullable : https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/api-guidelines/nullability.md#virtualabstract-methods-and-interfaces

**Comment by SteveWilkes** [comment]
That's really interesting, thank you! :)

**Comment by SteveWilkes** [comment]
This is fixed as of v3.3, which is now available [on NuGet](https://www.nuget.org/packages/AgileObjects.ReadableExpressions). Cheers!
